Add integration with Patchoulli jump-to-page feature

MrDrProfNo opened the pull request:
Addresses #168

Adds references to many entries in the docs to allow opening pages by shift-clicking a block with the book, or holding Ctrl on an item with the book in your inventory.

The .txt files used in making the book entries have had a third line inserted. This line is either empty, or a series of references + page numbers in the format

`ItemStack string[;page number][, ...]`

If no page number is provided, it defaults to 0.

All existing .txts have been updated to match this format.

My general philosophy in choosing these references is:
1. The block used as an entry's icon (and its variants) probably references that entry
2. Anything too generic (vanilla items, metals/gems, etc.) doesn't reference any page.
3.  ... best effort attempt to make anything not under 2. fit to a page.

It seems likely that I've missed _something_, but it should be fine for the most part.

I'll note that I've included the (updated) built docs files in this commit, but as generated files they should probably be removed from source control.

MrDrProfNo committed:
Add integration with Patchoulli jump-to-page feature

Da-Technomancer commented on src/main/resources/assets/crossroads/lang/en_us.json:
```diff
@@ -307,7 +307,7 @@
 	"item.minecraft.potion.effect.sedation": "Potion of Sedation",
 	"item.minecraft.splash_potion.effect.sedation": "Splash Potion of Sedation",
-	"item.minecraft.lingering_potion.effect.sedation": "Lingering Potion of Sedation",
+	"item.minecraft.lingering_potion.`effect.sedation": "Lingering Potion of Sedation",
```
Why the added backtick?

MrDrProfNo replied:
Yep, that was a mistake. Probably happened during an alt+tab. Sorry.

Fixed.

Da-Technomancer commented on src/main/resources/python/docs/template.txt:
```diff
@@ -7,5 +7,8 @@
-	]
+	],
+	"extra_recipe_mappings": {
+        MAPPINGS
```
The indenting on this line is off; see any of the generated files for why.

Da-Technomancer commented on src/main/resources/python/docs/template_adv.txt:
```diff
@@ -8,5 +8,8 @@
-	]
+	],
+	"extra_recipe_mappings": {
+        MAPPINGS
```
The indenting on this line is off; see any of the generated files for why.

MrDrProfNo replied:
Caught that pycharm decided to use 4 space indents on GenDocs.py, but not here. 

Fixed (in both this and template.txt)

Da-Technomancer commented on src/main/resources/python/docs/GenDocs.py:
```diff
@@ -90,17 +92,32 @@ def run():
+						continue
+
+					# page number within entry is optional, defaults 0
+					if ';' in mapping:
```
As an idea:

Considering that the script is supposed to handle splitting text into pages automatically, maybe the txt format shouldn't be based on setting a page number, as the page splits are generated by the script when it runs and usually aren't known in advance when writing the txt.

Instead, could the numbers refer to forced page breaks? (The things created by an empty line or <page> tag), so that when writing the txt, you would specify 'item_id;3' and the script would understand that as 'third page break explicitly specified by the txt' and write the correct page number in the json?

As a way to make it compatible with the work you've already done, perhaps have it be 'item_id;page_number' for page number the way you already have, and 'item_id$section_number' for an easier-to-write optional new way.

MrDrProfNo replied:
Updated based on discord discussion to instead have the numbers refer to the line which it should attempt to map the ItemStack string to, regardless of what page that line ends up on.

Updated... I'm _pretty sure_ all of the docs inputs accordingly.

Da-Technomancer reviewed (commented):
Mostly looks good.
A couple oddities in the formatting (I think a symbol got added to en_us.json by mistake?) and a suggestion.

MrDrProfNo committed:
Fix indentation on template files

MrDrProfNo committed:
Fix accidental grave character

MrDrProfNo committed:
Change method of linking to individual pages to go by start of line; push regenerated docs

Da-Technomancer committed:
Fixed incorrect fluid tube entry, clarified script header
*Fixed an incorrectly formatted entry for the fluid tube, which meant the script crashed when run
*Clarified in the script comment that it requires a new Python version and how it counts lines in the body
